examples/federated_learning/yolov5_coco128_mistnet/README.md

```diff
@@ -0,0 +1,238 @@
+MistNet is proposed to address this issue.
+
+MistNet partitions a DNN model into two parts, a lightweight feature extractor at the edge side to generate meaningful
+features from the raw data, and a classifier including the most model layers at the cloud to be iteratively trained for
+specific tasks. MistNet achieves acceptable model utility while greatly reducing privacy leakage from the released
+intermediate features.
+
+## Object Detection Experiment
+
+> Assume that there are two edge nodes and a cloud node. Data on the edge nodes cannot be migrated to the cloud due to privacy issues.
+> Base on this scenario, we will demonstrate the mnist example.
+
+### Prepare Nodes
+
+```
+CLOUD_NODE="cloud-node-name"
+EDGE1_NODE="edge1-node-name"
+EDGE2_NODE="edge2-node-name"
+```
+
+### Install Sedna
+
+Follow the [Sedna installation document](/docs/setup/install.md) to install Sedna.
+
+### Prepare Dataset
+
+Download [dataset](https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip) 
+
+```shell
+mkdir data
+wget https://github.com/ultralytics/yolov5/releases/download/v1.0/coco128.zip
+unzip coco128.zip -d data/COCO # the data path should be sedna/examples/federated_learning/yolov5_coco128_mistnet/data/COCO
+```
+
+Create data interface for ```EDGE1_NODE```.
+
+```
+mkdir -p /data
+echo "" > /data/test.txt
+```
+
+Create data interface for ```EDGE2_NODE```.
+
+```
+mkdir -p /data
+echo "" > /data/test.txt
+```
+
+### Prepare Images
+
+This example uses these images:
+
+1. aggregation worker: ```kubeedge/sedna-example-federated-learning-mistnet-yolo-aggregato:v0.4.0```
+2. train worker: ```kubeedge/sedna-example-federated-learning-mistnet-yolo-client:v0.4.0```
+
+These images are generated by the script [build_images.sh](/examples/build_image.sh).
+
+### Create Federated Learning Job
+
+#### Create Dataset
+
+create dataset for `$EDGE1_NODE` and `$EDGE2_NODE`
+
+```bash
+kubectl create -f - <<EOF
+apiVersion: sedna.io/v1alpha1
+kind: Dataset
+metadata:
+  name: "coco-dataset"
+spec:
+  url: "/data/test.txt"
+  format: "txt"
+  nodeName: $EDGE1_NODE
+EOF
+```
+
+```bash
+kubectl create -f - <<EOF
+apiVersion: sedna.io/v1alpha1
+kind: Dataset
+metadata:
+  name: "coco-dataset"
+spec:
+  url: "/data/test.txt"
+  format: "txt"
+  nodeName: $EDGE2_NODE
+EOF
+```
+
+#### Create Model
+
+create the directory `/model` in the host of `$CLOUD_NODE` (download links [here](https://kubeedge.obs.cn-north-1.myhuaweicloud.com/examples/yolov5_coco128_mistnet/yolov5.pth))
+
+```bash
+mkdir models
+mkdir models/pretrained/
+# put pretrained model on sedna/examples/federated_learning/yolov5_coco128_mistnet/models/pretrained/yolov5.pth
+```
+
+```
+mkdir -p /model
+echo "" > /model/yolo.pb
+```
+
+create model
+
+```bash
+kubectl create -f - <<EOF
+apiVersion: sedna.io/v1alpha1
+kind: Model
+metadata:
+  name: "yolo-v5-model"
+spec:
+  url: "/model/yolo.pb"
```

Comment:
why not `/model/yolo.th` ?
what is the use of `/model/yolo.th` ?

Reply:
updated